### PR TITLE
asn: init at 0.72.1

### PR DESCRIPTION
**Pull request by devhell**
###### Description of changes

Perhaps useful for #81418.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/asn/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+, jq
+, ipcalc
+, grepcidr
+, nmap
+, aha
+}:
+
+stdenv.mkDerivation rec {
+  version = "0.72.1";
+  pname = "asn";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/asn/default.nix:18]:
```suggestion
  pname = "asn";
  version = "0.72.1";
```

File: pkgs/applications/networking/asn/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  installPhase = ''
+    install -Dv asn "$out/bin/asn"
+
+    wrapProgram $out/bin/asn \
+      --prefix PATH : "${lib.makeBinPath [ curl whois bind mtr jq ipcalc grepcidr nmap aha ]}"
+  '';
+
+  meta = {
+    description = "ASN / RPKI validity / BGP stats / IPv4v6 / Prefix / URL / ASPath / Organization / IP reputation / IP geolocation / IP fingerprinting / Network recon / lookup API server / Web traceroute server";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/asn/default.nix:37]: That is a bit long for description. Can we shorten this and move the full list into longDescription?

Comment (devhell) [reply pkgs/applications/networking/asn/default.nix]: Agreed. I wasn't sure what to do with it, hope the change is better.

File: pkgs/applications/networking/asn/default.nix (hunk 1 above)
Comment (devhell) [reply pkgs/applications/networking/asn/default.nix]: Done.

File: pkgs/applications/networking/asn/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  installPhase = ''
+    install -Dv asn "$out/bin/asn"
+
+    wrapProgram $out/bin/asn \
+      --prefix PATH : "${lib.makeBinPath [ curl whois bind mtr jq ipcalc grepcidr nmap aha ]}"
+  '';
+
+  meta = {
+    description = "OSINT command line tool for investigating network data"
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/asn/default.nix:37]:
```suggestion
    description = "OSINT command line tool for investigating network data";
```

File: pkgs/applications/networking/asn/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    longDescription = ''
+      ASN / RPKI validity / BGP stats / IPv4v6 / Prefix / URL / ASPath / Organization / IP reputation / IP geolocation / IP fingerprinting / Network recon / lookup API server / Web traceroute server
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/asn/default.nix:39]: Please split this along multiple lines.

File: pkgs/applications/networking/asn/default.nix (hunk 3 above)
Comment (devhell) [reply pkgs/applications/networking/asn/default.nix]: Argh, sorry, completely missed that.

File: pkgs/applications/networking/asn/default.nix (hunk 4 above)
Comment (devhell) [reply pkgs/applications/networking/asn/default.nix]: I wasn't sure, so I left it originally. I've split it up now.

**Pull request merged**